snprc_scheduler/src/org/labkey/snprc_scheduler/domains/Timeline.java

```diff
@@ -118,7 +120,8 @@ public Timeline(Container c, User u, JSONObject json) throws RuntimeException
             this.setCreatedByName(json.optString(TIMELINE_CREATED_BY_NAME, null));
             this.setModifiedByName(json.optString(TIMELINE_MODIFIED_BY_NAME, null));
             this.setDescription(json.optString(TIMELINE_DESCRIPTION, null));
-            this.setQcState(json.isNull(TIMELINE_QCSTATE) ? null : json.getInt(TIMELINE_QCSTATE));
+            // update QCState from the QCStateLabel field
+            this.setQcState(json.isNull(TIMELINE_QCSTATE_LABEL) ? null : QCStateEnum.getValueByName (c, u, QCStateEnum.getByName(json.optString(TIMELINE_QCSTATE_LABEL))));
```

Comment:
null check for getByName()

Reply:
Null is allowed - that is why I used optString() instead of getString() - The validation function will set the default value for the QcState if it is null.